src/main/java/de/tum/in/www1/artemis/web/rest/AbstractSubmissionResource.java

```diff
@@ -0,0 +1,90 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import java.util.List;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.repository.SubmissionRepository;
+import de.tum.in.www1.artemis.service.*;
+import de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException;
+
+/**
+ * REST controller for managing Submission.
+ */
+@RestController
+@RequestMapping("/api")
+public class AbstractSubmissionResource {
```

Comment:
Should be `abstract`.

Reply:
Also, I would rename this to `SubmissionResource` and remove `@Annotations` to be consistent with the `AssessmentResource` class.

Reply:
Good point. I made it abstract. The issue is that there is already a generic SubmissionResource with API endpoints and it was not possible to apply inheritance here, otherwise those API endpoints would be inherited by the subclasses and would be available twice. There we now have a generic `SubmissionResource` (offers generic API endpoints for REST calls that are independent of the exercise type) and an abstract `AbstractSubmissionResource` which is used as superclass to avoid duplicated code.

I am not really happy with this solution, but I think we should improve this structure further in a follow-up PR.